xtask/src/main.rs

```diff
@@ -15,6 +15,50 @@ use crate::utils::{
 
 static ALL_ERRORS: Lazy<Mutex<Vec<String>>> = Lazy::new(|| Mutex::new(vec![]));
 
+const ALL_SNAPSHOT_TESTS: [&str; 12] = [
+    "log",
+    "bitflags",
+    "timestamp",
+    "panic",
+    "assert",
+    "assert-eq",
+    "assert-ne",
+    "unwrap",
+    "defmt-test",
+    "hints",
+    "hints_inner",
+    "dbg",
+];
+
+#[derive(Debug, StructOpt)]
+#[allow(clippy::enum_variant_names)]
+enum Snapshot {
+    All,
+    Single { name: String },
+}
+
+impl FromStr for Snapshot {
+    type Err = String;
+
+    fn from_str(source: &str) -> Result<Self, Self::Err> {
+        match source {
+            "all" => Ok(Snapshot::All),
+            test => {
+                if ALL_SNAPSHOT_TESTS.contains(&test) {
```

Comment:
This doesn't work with the conditionally-available `alloc` test, but that's probably not a big problem